File: app/src/main/java/com/todolist/main/BtnClickTaskListener.java
package com.todolist.main;

import android.view.View;

import com.todolist.db.TaskTableHelper;
import com.todolist.models.Task;

public class BtnClickTaskListener implements View.OnClickListener {
    private Task task;
    private TaskTableHelper taskTableHelper;
    private MainActivity mainActivity;


    public BtnClickTaskListener(Task task, TaskTableHelper taskTableHelper, MainActivity mainActivity) {
        this.task = task;
        this.taskTableHelper = taskTableHelper;
        this.mainActivity = mainActivity;
    }

    @Override
    public void onClick(View view) {
        taskTableHelper.delete(task);
        mainActivity.RefreshListView();
    }
}

File: Study/ToDoList/app/src/main/java/ru/brovkin/todolist/main/BtnClickTaskListener.kt
package ru.brovkin.todolist.main

import android.view.View
import ru.brovkin.todolist.models.Task
import ru.brovkin.todolist.repositories.TaskRepository

class BtnClickTaskListener : View.OnClickListener{
    private lateinit var  task : Task
    private lateinit var taskRepository: TaskRepository
    private lateinit var mainActivity: MainActivity

    constructor(task: Task, taskRepository: TaskRepository, mainActivity: MainActivity) {
        this.task = task
        this.taskRepository = taskRepository
        this.mainActivity = mainActivity
    }

    override fun onClick(view: View) {
        taskRepository.delete(task)
        mainActivity.RefreshListView()
    }
}